User Retention Feature for Security Release (#1343)
rancher/rancher-docs: docs/how-to-guides/advanced-user-guides/enable-user-retention.md

> ---

Reviewer: Sidebar entry is missing for this file.

title: Enabling User Retention
---

<head>

>   <link rel="canonical" href="https://ranchermanager.docs.rancher.com/how-to-guides/advanced-user-guides/enable-user-retention.md"/>

Author: Fixed in 51e1d23b77361e5025ec718b02f2819cbb8855b3 alongside other instances

</head>

> In Rancher v2.8.5 and later (and v2.7.14 and later in the v2.7.x line), you can enable user retention to automatically disable or delete inactive user accounts after a configurable time period.

Reviewer: ```suggestion
In Rancher v2.8.5 and later, you can enable user retention to automatically disable or delete inactive user accounts after a configurable time period.
```

We should minimize cross-version references since users shouldn't reading version X docs aren't going to be concerned about version Y (especially if it's older).

The user retention feature is off by default.

> ## Enabling User Retention with Kubectl

Reviewer: ```suggestion
## Enabling User Retention with kubectl
```

To enable user retention, use kubectl to set `user-retention-cron`. Then, set `delete-inactive-user-after`, `disable-inactive-user-after`, or a combination of both:

```
kubectl edit setting user-retention-cron 

> kubectl edit setting disable-inactive-user-after

Reviewer: The `### Required User Retention Settings` section mentions these expect certain values, but that's not indicated here.

Author: Originally, this was a placeholder for a working example, but the values were removed in https://github.com/rancher/rancher-docs/pull/1343/commits/2f4049e995c8d3ef24a3381282887d40ac774bc1 without clarifying. It might be better to state "use `kubectl edit setting` to ..." instead of providing the snippet

```

## Configuring Rancher to Delete Users, Disable Users, or Combine Operations

> Rancher uses two global user retention settings to determine if and when users are disabled or deleted after a certain period of inactivity. Disabled accounts must be re-enabled before users can login again. If an account is deleted without being disabled, users may be able to login through external authentication and the deleted account will be recreated.

Reviewer: ```suggestion
Rancher uses two global user retention settings to determine if and when users are disabled or deleted after a certain period of inactivity. Disabled accounts must be re-enabled before users can log in again. If an account is deleted without being disabled, users may be able to log in through external authentication and the deleted account will be recreated.
```

The global settings, `disable-inactive-user-after` and  `delete-inactive-user-after`, do not block one another from running. 

For example, you can set both operations to run. If you give `disable-inactive-user-after` a shorter duration than `delete-inactive-user-after`, the user retention process disables inactive accounts before deleting them.

You can also edit some user retention settings on a specific user's `UserAttribute`. Setting these values overrides the global settings. See [User-specific User Retention Overrides](#user-specific-user-retention-overrides) for more details.

### Required User Retention Settings

The following are global settings:

- `user-retention-cron`: Describes how often the user retention process runs. The value is a cron expression (for example, `0 * * * *` for every hour).

> - `disable-inactive-user-after`: The amount of time that a user account can be inactive before the process disables an account. Disabling an account forces the user to request that an administrator re-enable the account before they can login to use it. Values are expressed in [time.Duration units](https://pkg.go.dev/time#ParseDuration) (for example, `720h` for 720 hours or 30 days). The value must be greater than `auth-user-session-ttl-minutes`, which is `16h` by default. If the value is not set, set to the empty string, or is equal to 0, the process does not disable any inactive accounts.

Reviewer: ```suggestion
- `disable-inactive-user-after`: The amount of time that a user account can be inactive before the process disables an account. Disabling an account forces the user to request that an administrator re-enable the account before they can log in to use it. Values are expressed in [time.Duration units](https://pkg.go.dev/time#ParseDuration) (for example, `720h` for 720 hours or 30 days). The value must be greater than `auth-user-session-ttl-minutes`, which is `16h` by default. If the value is not set, set to the empty string, or is equal to 0, the process does not disable any inactive accounts.
```

- `delete-inactive-user-after`: The amount of time that a user account can be inactive before the process deletes the account. Values are expressed in time.Duration units (for example, `720h` for 720 hours or 30 days). The value must be greater than `auth-user-session-ttl-minutes`, which is `16h` by default. The value should be greater than `336h` (14 days), otherwise it is rejected by the Rancher webhook. If you need the value to be lower than 14 days, you can [bypass the webhook](../../reference-guides/rancher-webhook.md#bypassing-the-webhook). If the value is not set, set to the empty string, or is equal to 0, the process does not delete any inactive accounts.

> ::: important

Reviewer: ```suggestion
:::important
```

> To enable user retention, you must set `user-retention-cron`. You must also set at least one of  `disable-inactive-user-after` or `delete-inactive-user-after`.

Reviewer: This note seems a bit repetitive since the section itself is about required settings and the only options in the section are these 3. Thoughts on removing the note and moving the stronger language (e.g. `must`) from this note to the `# Enabling...` section where it's more relevant?

---
title: Enabling User Retention
---

<head>
  <link rel="canonical" href="https://ranchermanager.docs.rancher.com/how-to-guides/advanced-user-guides/enable-user-retention"/>
</head>

In Rancher v2.8.5 and later, you can enable user retention to automatically disable or delete inactive user accounts after a configurable time period.

The user retention feature is off by default.

>  ## Enabling User Retention with kubectl

Reviewer: ```suggestion
## Enabling User Retention with kubectl
```
